maven-repository-manager/src/main/java/org/jboss/pnc/mavenrepositorymanager/RepositoryManagerDriver.java

```diff
@@ -270,14 +276,27 @@ private void setupBuildRepos(BuildExecution execution, Indy indy)
             // Global-level repos, for captured/shared artifacts and access to the outside world
             addGlobalConstituents(buildGroup, tempBuild);
 
-            // add extra repositories removed from poms by the adjust process
-            addExtraConstituents(execution.getArtifactRepositories(), id, buildContentId, indy, buildGroup);
+            // add extra repositories removed from poms by the adjust process and set in BC by user
+            List<ArtifactRepository> extraDependencyRepositories = extractExtraRepositoriesFromGenericParameters(genericParameters);
+            extraDependencyRepositories.addAll(execution.getArtifactRepositories());
+            addExtraConstituents(extraDependencyRepositories, id, buildContentId, indy, buildGroup);
 
             indy.stores().create(buildGroup, "Creating repository group for resolving artifacts in build: " + id
                     + " (repo: " + buildContentId + ")", Group.class);
         }
     }
 
+    private List<ArtifactRepository> extractExtraRepositoriesFromGenericParameters(Map<String, String> genericParameters) {
+        String extraReposString = genericParameters.get(EXTRA_PUBLIC_REPOSITORIES_KEY);
+        if (extraReposString == null)
+            return Collections.emptyList();
+
+        return Arrays.stream(extraReposString.split("\n"))
+                .map(repoString ->
+                        ArtifactRepository.build(repoString.trim(), repoString.trim(), repoString.trim(), true, false))
```

Comment:
Could we introduce some format here including also an id? Something like
```central http://repo1.maven.org/maven2/```
If id not specified just convert the hostname by replacing dots for hyphens, e.g.
```repo1-maven-org```
The value can be used for both id and name to not complicate it any more.

Reply:
I'll go for the generation as you proposed. IMHO less input from users better UX